docs/DeveloperGuide.md

```diff
@@ -1,38 +1,558 @@
+
+- **Online Resources & Developer Communities**
+    - We also appreciate the vast knowledge shared by developers on platforms like Stack Overflow, GitHub, and various Java documentation sources that helped us troubleshoot and enhance our implementation.
+
+
+---
+
+<h2>
+<span style="color:orange;">Setting up & Getting Started</span>
+</h2>
+
+<div style="background-color: #fff3cd; border-left: 6px solid #ffa502; padding: 10px;">
+<strong>⚠️  Caution: Follow the steps in the following guide precisely. </strong>
+The code may not work as intended if steps are deviated from
+</div>
+
+<br/>
+
+First, **fork** this repo, and **clone** the fork into your computer.
+
+##### If you plan to use Intellij IDEA (highly recommended):
+
+**Configure the JDK:** Follow the guide [se-edu/guides] IDEA: Configuring the JDK to ensure Intellij is configured to use JDK 17.
+
+**Import the project as a Gradle project:** For instructions on importing a Gradle project into IntelliJ IDEA, refer to the [se-edu guide](https://se-education.org/guides/tutorials/intellijImportGradleProject.html).
+
+
+<div style="background-color: #89CFF0; border-left: 6px solid #ffa502; padding: 10px;">
+<strong> Note: </strong> Importing a Gradle project is slightly different from importing a normal Java project.
+</div>
+<br/>
+
+**Verify the setup:**
+- Run the seedu.address.Main and try a few commands.
+- Run the tests to ensure they all pass.
+- Refer to the **Manual Testing** section for more in-depth instructions on testing
+
+
+
+---
+
+<h2>
+<span style="color:orange;">Design</span>
+</h2>
+
+<h4>
+<span style="color:orange;">Architechture</span>
+</h4>
+
+
+![image](diagrams/Architecture.png)
+
+The **_Architecture Diagram_** given above explains the high-level design of
+the App.
+
+Given below is a quick overview of main components and how they interact with
+each other.
+
+**Main components of the architecture**
+
+**`GrandRhombus`** is in charge of the app launch and shut down.
+
+- At app launch, it initializes the other components in the correct sequence, and connects them up with each other.
+- At shut down, it shuts down the other components and invokes cleanup methods where necessary.
+
+The bulk of the app’s work is done by the following three components:
+
+- **`CommandParser`**: Parses user input and chooses the command
+- **`Command`**: The individual commands which are then executed. 
+- **`Data`**: A collection of classes including `User`, `UserData` which loads and saves user's modules (`Mod`) into **`user.txt`** file
+
+*`Resources`* contains the `moduledata.txt` file which serves as the database of all CEG modules. 
+These modules are converted into `Mod`s to be stored in other components.
+
+**How the architecture components interact with each other**
+
+The _Sequence Diagram_ below shows how the components interact with each other from startup to termination.
+![image](diagrams/SequenceDiagram.png)
```

Comment:
The class names in the diagram are actually objects (except that we dont know the object name so we leave it blank) but we put a colon infront of the class names. The colons are missing in the diagram so please add them.